packages/mui-joy/src/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,789 @@
+    nameOrDescProps['aria-describedby'] = open ? id : null;
+  } else {
+    nameOrDescProps['aria-label'] = titleIsString ? title : null;
+    nameOrDescProps['aria-labelledby'] = open && !titleIsString ? id : null;
+  }
+
+  const childrenProps = {
+    ...nameOrDescProps,
+    ...other,
+    ...children.props,
+    className: clsx(className, children.props.className),
+    onTouchStart: detectTouchStart,
+    ref: handleRef,
+    ...(followCursor ? { onMouseMove: handleMouseMove } : {}),
+  };
+
+  const interactiveWrapperListeners: {
+    onMouseOver?: (event: React.MouseEvent<HTMLElement>) => void;
+    onMouseLeave?: (event: React.MouseEvent<HTMLElement>) => void;
+    onFocus?: (event: React.FocusEvent<HTMLElement>) => void;
+    onBlur?: (event: React.FocusEvent<HTMLElement>) => void;
+  } = {};
+
+  if (!disableTouchListener) {
+    childrenProps.onTouchStart = handleTouchStart;
+    childrenProps.onTouchEnd = handleTouchEnd;
+  }
+
+  if (!disableHoverListener) {
+    childrenProps.onMouseOver = composeMouseEventHandler(
+      handleMouseOver,
+      childrenProps.onMouseOver,
+    );
+
+    childrenProps.onMouseLeave = composeMouseEventHandler(
+      handleMouseLeave,
+      childrenProps.onMouseLeave,
+    );
+
+    if (!disableInteractive) {
+      interactiveWrapperListeners.onMouseOver = handleMouseOver;
+      interactiveWrapperListeners.onMouseLeave = handleMouseLeave;
+    }
+  }
+
+  if (!disableFocusListener) {
+    childrenProps.onFocus = composeFocusEventHandler(handleFocus, childrenProps.onFocus);
+    childrenProps.onBlur = composeFocusEventHandler(handleBlur, childrenProps.onBlur);
+
+    if (!disableInteractive) {
+      interactiveWrapperListeners.onFocus = handleFocus;
+      interactiveWrapperListeners.onBlur = handleBlur;
+    }
+  }
+
+  const popperOptions = React.useMemo(() => {
+    let tooltipModifiers = [
+      {
+        name: 'arrow',
+        enabled: Boolean(arrowRef),
+        options: {
+          element: arrowRef,
+          // https://popper.js.org/docs/v2/modifiers/arrow/#padding
+          // make the arrow looks nice with the Tooltip's border radius
+          padding: 6,
+        },
+      },
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 10],
+        },
+      },
+    ];
+
+    if (RootProps.popperOptions?.modifiers) {
+      tooltipModifiers = tooltipModifiers.concat(RootProps.popperOptions.modifiers);
+    }
+
+    return {
+      ...RootProps.popperOptions,
+      modifiers: tooltipModifiers,
+    };
+  }, [arrowRef, RootProps]);
+
+  const ownerState = {
+    ...props,
+    arrow,
+    disableInteractive,
+    placement,
+    RootComponentProp,
+    touch: ignoreNonTouchEvents.current,
+    color,
+    variant,
+    size,
+  };
+
+  const classes = useUtilityClasses(ownerState);
+
+  const RootComponent = components.Root ?? TooltipRoot;
+  const ArrowComponent = components.Arrow ?? TooltipArrow;
+
+  const popperProps = appendOwnerState(
```

Comment:
This should use `useSlotProps` util from MUI Base. Please take a look at `Select` component for example.